modules/saveload/register_types.cpp

```diff
@@ -0,0 +1,75 @@
+/**************************************************************************/
+/*  register_types.cpp                                                    */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "register_types.h"
+
+#include "saveload_api.h"
+#include "saveload_spawner.h"
+#include "saveload_synchronizer.h"
+#include "scene_saveload.h"
+#include "scene_saveload_interface.h"
+
+#include "saveload_debugger.h"
+
+#ifdef TOOLS_ENABLED
+#include "editor/saveload_editor_plugin.h"
+#endif
+
+static SceneSaveload *saveload_api = NULL;
+
+void initialize_saveload_module(ModuleInitializationLevel p_level) {
+	if (p_level == MODULE_INITIALIZATION_LEVEL_SERVERS) {
+		saveload_api = memnew(SceneSaveload);
+		GDREGISTER_ABSTRACT_CLASS(SaveloadAPI);
+		GDREGISTER_ABSTRACT_CLASS(SceneSaveload);
+		Engine::get_singleton()->add_singleton(Engine::Singleton("SceneSaveload", saveload_api));
```

Comment:
```suggestion
		Engine::get_singleton()->add_singleton(Engine::Singleton("SaveloadAPI", SaveloadAPI::get_singleton(), "SaveloadAPI"));
```
The unit tests require that singletons be derived directly from `Object`, and you're using different names for the singletons here and when unregistering

Reply:
I'm testing these changes locally and running the CI on my repository just to check

Reply:
Rebasing it to make building more convenient is good, but remember to rebase and not merge in the future for updating the branch

You need to add documentation specifically for the module classes, try building and running `--doctool`

Reply:
Curses! The .NET test still isn't happy it looks like. I'm guessing it doesn't like that I called the singleton "SaveloadAPI". I'll try renaming it to "SceneSaveload" to see if that makes it happy.

Reply:
You might have to unregister `ScendSaveload` as it seems to not like to exist in C# since it derives from a static type, try that as a solution (remember to remove the documentation as well)

Reply:
I could also try not exposing `SceneSaveload` for scripting.

Reply:
Jinx 🙃

Reply:
Do I also need to remove the
```cpp
GDCLASS(SceneSaveload, SaveloadAPI);
```
in `SceneSaveload.h`, or just `GDREGISTER_ABSTRACT_CLASS(SceneSaveload);` in `register_types.cpp`?

Reply:
Pushed it! I made a little Git oopsie and accidentally pushed two different commits. Just run the CI on the latest one.

Reply:
Roger that!

Final run, let's gooooooooo! :D

Reply:
Seems like it passed the .NET tests :D 

The logs for editor w/ doubles confuse me. It seems to be failing on something completely unrelated to saveload.

Reply:
I see, that explains the multiple attempts. Thanks for the info!

You can do it little builder! 🙌